### DISCUSSION TITLE
Integracja sketchu z  Home Assistant

### DISCUSSION THREAD

**Discussion by paku2020** [lkankowski/arduino-multi-relay2#33] [General]
Dzień dobry :) 
Chciałbym zapytać czy sketch który działa poprawnie na domoticznu będzie również działał w HA  czy może trzeba coś w min zmienić?
Chciałbym przenieść to do Home Assistenta lecz coś mi nie trybi i zastanawiam się w czym może być problem.

Pozdrawiam

**Comment by pepsonEL** [comment]
Witam
Używam Sketcha Leszka już od dawna w HA i działa ok. Używam obecnie najnowszej wersji....

**Reply by paku2020** [reply]
Hej.
Dzięki za odpowiedz.
A coś trzeba zmienić w wersji dev aby działało pod HA?

**Comment by pepsonEL** [comment]
Szczerze już nie pamiętam ale weź z mastera sketch....

**Reply by paku2020** [reply]
Zależy mi właśnie na obsłudze przez lan.
   W HA tworzy mi plik  /config/mysensors_f043836cc3a57f52fef86e7b71242620.json   ale w logach mam błąd 
![image](https://user-images.githubusercontent.com/62469055/148912495-9b85b7cb-b1fd-4bfe-bfa7-379dc8fdaf94.png)

oraz 
![image](https://user-images.githubusercontent.com/62469055/148912526-833762f1-2d2d-4ede-92ba-0bf12c193db3.png)

Na domoticzmu mi działa to poprawnie nawet, ale chcę przejść na HA i zaczynają się schody :)

**Reply by pepsonEL** [reply]
Ja używam przez USB bo problem w LAN jest taki że jeśli nie działa LAN to Sketch z Arduino też nie działa. Na USB jest o tyle plus że wystarczy że ma zasilanie z USB i działa.... Pada Ci np HA dajesz zwykłą ładowarkę od tel pod USB i zasilasz Arduino i światło działa poprawnie.  LAN wiem że jest niedopracowany w MySensors. A takie ostrzeżenia też mam ale nie przejmuje się nimi.

**Reply by paku2020** [reply]
`A takie ostrzeżenia też mam ale nie przejmuje się nimi.`
W  twoim przypadku działa :)
La wybrałem lan ponieważ mam wyniesioną bramkę. Też walczylem z tym i teraz chyba jest oki, trzeba tylko zrobić zasilanie chyba po usb właśnie i ip na sztywno wpisać

**Reply by pepsonEL** [reply]
Aha może i tak.  Ja już mam na USB i na tym zostaję....

**Comment by paku2020** [comment]
A czy przypadkiem  nie trzeba zdefiniować
`#define MY_NODE_ID 100`
?

**Comment by pepsonEL** [comment]
Chyba nie przynajmniej autor nic o tym nie wspominał....

wt., 11 sty 2022, 16:32 użytkownik paku2020 ***@***.***>
napisał:

**Reply by lkankowski** [reply]
Nie słuchaj autora :P
Wg dokumentacji (https://www.mysensors.org/download/sensor_api_20) domyślna wartość MY_NODE_ID=AUTO, więc faktycznie nie trzeba.
Natomiast jeszcze raz przypominam, że obsługa LAN/MQTT jest EKSPERYMENTALNA!!!
Znaczy to tyle, że poza zgróbnymi testami, robieniem wg dokumentacji i na wyczucie, ja nic więcej nie testowałem. Nie korzystam u siebie i naprawdę nie mam czasu, aby takie dogłębne testy przeprowadzać.

Co do wcześniejszych komentarzy:
1. Warningi "DEPRECATED" - ogólnie radzę na nie uważać, bo po którejś aktualizacji HA dadzą kopa w d... (choć w tym wypadku to niegroźne)
2. HA ma słabą obsługę MySensors - np. jeden z "niewyjaśnionych" problemów mojego sketcha, czyli masowe wyłączanie wszystkich urządzeń ostatnio robiłem na Domoticzu i działa wyśmienicie :)  Nie wiem czy to jakiś efekt uboczny i Domoticz robi to tak wolno, ale światła fajnie gasną pomieszczenie po pomieszczeniu jakby było jakieś opóźnienie. HA "na wariata" wysyła wszystko na UART, a przecież wiadomo, że to wolny interfejs i nie przeznaczony do takiego wolumenu danych
3. Ogólnie MySensors to fajny protokół, ale... do małych radiowych sensorów, po kilka per gateway, a nie po 30-60 urządzeń. Mam w planach zrobić wariant MQTT bez MySensors ale za to z Autodiscovery - oczywiście nie mam wielkiej motywacji, bo w dalszym ciągu nie będę tego używać :) - chyba jedynie dla "fejmu" :D

**Reply by pepsonEL** [reply]
Jak zrobisz po MQTT z Autodiscovery to jestem za i jestem chętny jako pierwszy BETA TESTER tym bardziej z ok 60 urządzeniami na jednym ArduinoMega z Ekspanderem MCP23017 64 piny.

**Reply by pepsonEL** [reply]
Ale szczerze fajnie jakby udało się przepisać ten sketch na MQTT z Autodiscovery. Jak coś dawaj znać :)

**Comment by paku2020** [comment]
Panowie, a gdybym chciał dodać do kodu
`#define MY_NODE_ID 100`
to w którym miejscu należy to dopisać ? 
Chciałbym to przetestować, może w moim przypadku w tym jest jakiś problem

**Reply by lkankowski** [reply]
Musi być przed `#include <MySensors.h>`, a więc najlepiej w `config.h`.

**Reply by paku2020** [reply]
A w jakimś konkretnym miejscu czy na w ostatniej linii jak wpisze to będzie ok ?

**Reply by lkankowski** [reply]
Spróbuj i powiedz jaki był efekt. Bez obaw, niczego nie zepsujesz.

**Reply by paku2020** [reply]
Wydaje mi się że cały czas jest tak jakby ten wpis nic nie zmieniał.   Dalej jest node 0 w HA.
Zauważyłem też dwie rzeczy, jedna to chyba już wiadomo, trzeba ręcznie dopisać 
```
                "values": {
                    "2": "0"
                }
```
Po dodaniu bramki mysensors  w konfiguracji bramki jest 
```
                "values": { }
```
Druga sprawa to testuje ten sam sketch na  arduino mega +W5100  oraz na wemos d1 mini. Na wemosie po dodaniu w/w linijki chyba jest ok. Na arduino coś nie chce, tak jakby był problem z komunikacją. Jeszcze jutro sprawdzę na mega raz
